pyface/color.py

```diff
@@ -0,0 +1,260 @@
+
+
+class Color(HasStrictTraits):
+    """ A mutable specification of a color with alpha.
+
+    This is a class designed to be used by user interface elements which
+    need to color some or all of the interface element.  Each color has a
+    number of different representations as channel tuples, each channel
+    holding a value between 0.0 and 1.0, inclusive.  The standard red,
+    green, blue and alpha channels are also provided as convenience
+    properties.
+
+    Methods are provided to convert to and from toolkit-specific color
+    objects.
+
+    Colors implement equality testing, but are not hashable as they are
+    mutable, and so are not suitable for use as dictionary keys.  If you
+    need a dictionary key, use an appropriate channel tuple from the
+    object.
+    """
+
+    #: A tuple holding the red, green, blue, and alpha channels.
+    rgba = AlphaChannelTuple()
+
+    #: A tuple holding the red, green, and blue channels.
+    rgb = Property(ChannelTuple(), depends_on='rgba')
+
+    #: The red channel.
+    red = Property(Channel, depends_on='rgba')
+
+    #: The green channel.
+    green = Property(Channel, depends_on='rgba')
+
+    #: The blue channel.
+    blue = Property(Channel, depends_on='rgba')
+
+    #: The alpha channel.
+    alpha = Property(Channel, depends_on='rgba')
+
+    #: A tuple holding the hue, saturation, value, and alpha channels.
+    hsva = Property(AlphaChannelTuple, depends_on=['rgba'])
+
+    #: A tuple holding the hue, saturation, and value channels.
+    hsv = Property(ChannelTuple, depends_on=['rgb'])
+
+    #: A tuple holding the hue, lightness, saturation, and alpha channels.
+    hlsa = Property(AlphaChannelTuple, depends_on=['rgba'])
+
+    #: A tuple holding the hue, lightness, and saturation channels.
+    hls = Property(ChannelTuple, depends_on=['rgb'])
+
+    @classmethod
+    def from_toolkit(cls, toolkit_color, **traits):
+        """ Create a new Color object from a toolkit color object.
+
+        Parameters
+        ----------
+        toolkit_color : toolkit object
+            A toolkit color object, such as a Qt QColor or a Wx wx.Colour.
+        **traits
+            Any additional trait values to be passed as keyword arguments.
+        """
+        from pyface.toolkit import toolkit_object
+        toolkit_color_to_rgba = toolkit_object('color:toolkit_color_to_rgba')
+        rgba = toolkit_color_to_rgba(toolkit_color)
+        return cls(rgba=rgba, **traits)
+
+    def to_toolkit(self):
+        """ Create a new toolkit color object from a Color object.
+
+        Returns
+        -------
+        toolkit_color : toolkit object
+            A toolkit color object, such as a Qt QColor or a Wx wx.Colour.
+        """
+        from pyface.toolkit import toolkit_object
+        rgba_to_toolkit_color = toolkit_object('color:rgba_to_toolkit_color')
+        return rgba_to_toolkit_color(self.rgba)
+
+    def hex(self):
```

Comment:
Q : Why not have this as a property as well?

Reply:
Mainly because it isn't settable.  It could be, but that would probably be better with the parsing code.

Reply:
fair enough. resolving.